lib/features/common/data/repository/refresh_tokens_repository.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:async';
+
+import 'package:dio/dio.dart';
+import 'package:flutter_template/features/common/data/repository/api_caller_repository_mixin.dart';
+import 'package:flutter_template/features/common/domain/entity/operation/failure.dart';
+import 'package:flutter_template/features/common/domain/entity/operation/request_operation.dart';
+import 'package:flutter_template/features/common/domain/entity/operation/result.dart';
+import 'package:flutter_template/features/common/domain/entity/tokens/tokens_entity.dart';
+import 'package:flutter_template/features/common/domain/repository/refresh_tokens_repository.dart';
+import 'package:flutter_template/persistence/storage/tokens_storage/i_tokens_storage.dart';
+
+/// {@template refresh_tokens_repository.class}
+/// Implementation RefreshTokensRepository.
+/// {@endtemplate}
+final class RefreshTokensRepository
+    with ApiCallerRepositoryMixin
+    implements IRefreshTokensRepository {
+  final ITokensStorage _tokensStorage;
+
+  // TODO(anyone): implement
+  /// Add an Api for working with authorization tokens
+  /// Add a converter for mapping data from the data layer to the domain layer. Use `Converter`.
+
+  /// To control parallel token refresh
+  Completer<RequestOperationCompleter<TokensEntity>>? _refreshCompleter;
+
+  /// {@macro refresh_tokens_repository.class}
+  RefreshTokensRepository({
+    required ITokensStorage tokensStorage,
+  }) : _tokensStorage = tokensStorage;
+
+  @override
+  RequestOperation<TokensEntity> refreshTokens(String refreshToken) async {
+    if (_refreshCompleter != null) {
+      final result = await _refreshCompleter!.future;
+
+      switch (result) {
+        case ResultOk(:final data):
+          return Result.ok(data);
+        case ResultFailed(:final failure):
+          return Result.failed(failure);
+      }
+    }
+
+    _refreshCompleter = Completer();
+
+    try {
+      // TODO(anyone): implement
```

Comment:
Can we make this cleaner in some way? It's not very clear what this comment refers to and how to work with it.